pkg/apis/autoscaling/register.go

```diff
@@ -52,9 +52,42 @@ const (
 	// TargetAnnotationKey is the annotation to specify what metric value the
 	// PodAutoscaler should attempt to maintain. For example,
 	//   autoscaling.knative.dev/metric: cpu
-	//   autoscaling.knative.dev/target: 75   # target 75% cpu utilization
+	//   autoscaling.knative.dev/target: "75"   # target 75% cpu utilization
 	TargetAnnotationKey = GroupName + "/target"
 
+	// WindowAnnotationKey is the annotation to specify the time
+	// interval over which to calculate the average metric.  Larger
+	// values result in more smoothing. For example,
+	//   autoscaling.knative.dev/metric: concurrency
+	//   autoscaling.knative.dev/window: "2m"
+	// Only the kpa.autoscaling.knative.dev class autoscaler supports
+	// the window annotation.
+	WindowAnnotationKey = GroupName + "/window"
+
+	// WindowPanicPercentageAnnotationKey is the annotation to
+	// specify the time interval over which to calculate the average
+	// metric during a spike. Where a spike is defined as the metric
+	// reaching panic level within the panic window (e.g. panic
+	// mode). Lower values make panic mode more sensitive. Note:
+	// Panic level can be overridden with the
+	// TargetPanicPercentageAnnotationKey. For example,
+	//   autoscaling.knative.dev/windowPanicPercentage: "5.0"
+	//   autoscaling.knative.dev/targetPanicPercentage: "150.0"
```

Comment:
Validation will not allow this example. The max should be 100.

Reply:
Actually `targetPanicPercentage` should be more than 100%.  Currently it's hardcoded at 200% (2x).  That means the threshold to panic is 2x the target threshold.

![image](https://user-images.githubusercontent.com/1103629/52449432-3b79d780-2aec-11e9-9f9d-38038eb85578.png)

Reply:
Uargh, sorry. I misread this to be windowPanicPercentage as well. Nevermind.